packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2300,12 +2300,50 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     }
   }
 
+  Set<SemanticsAction> _mutedSemanticsActions = <SemanticsAction>{};
+
+  /// Update the set of [SemanticsAction] that will be ignore when constructing
+  /// semantics action handlers.
+  ///
+  /// The muted [SemanticsAction] will apply to all the children in subtree.
+  void updateMutedSemanticsActions(Set<SemanticsAction> actions) {
+    assert(actions != null);
+    if (_mutedSemanticsActions.length == actions.length &&
+        _mutedSemanticsActions.every(actions.contains))
+      return;
+    _mutedSemanticsActions = actions;
+    markNeedsSemanticsUpdate();
+    bool markRenderObjectNeedsSemanticsUpdate(RenderObject renderChild) {
+      renderChild.markNeedsSemanticsUpdate();
+      renderChild.visitChildrenForSemantics(markRenderObjectNeedsSemanticsUpdate);
+    }
+    visitChildrenForSemantics(markRenderObjectNeedsSemanticsUpdate);
+  }
+
+  // Use [_inheritedMutedSemanticsActions] to access
+  Set<SemanticsAction> _cachedMutedSemanticsActions;
+
+  /// The muted [SemanticsAction] that aggregated from parent and current render
+  /// object.
+  Set<SemanticsAction> get _inheritedMutedSemanticsActions {
+    if (_cachedMutedSemanticsActions == null) {
```

Comment:
How is the cache invalidated if the parent's _inheritedMutedSemanticsActions change?

Reply:
the recursive markRenderObjectNeedsSemanticsUpdate should invalidate all the subtree cache.

Reply:
That is a departure from how SemanticsUpdate marking works right now. Right now, if node A is marked as needing an update, all its children will also be updated without requiring to be marked as needing an update.

The subtree could potentially be very large (think about a modal dialog that blocks interactions with everything in the route below it), so performance-wise it may not be a good idea to walk everything.